plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php

```diff
@@ -0,0 +1,542 @@
+<?php
+/**
+ * @package plugins.CortexApiDistribution
+ * @subpackage lib
+ */
+class CortexApiDistributionEngine extends DistributionEngine implements
+	IDistributionEngineSubmit,
+	IDistributionEngineDelete
+{
+	const CORTEX_API_LOGIN = 'API/Authentication/v1.0/Login';
+	const CORTEX_API_SEND_METADATA = 'API/v2.2/DataTable/Documents.Video.General-Library-Video:Update';
+	const CORTEX_API_UPLOAD_NEW_MEDIA = 'API/UploadMedia/v3.0/UploadNewMedia';
+	const CORTEX_API_UPLOAD_CAPTIONS = 'webapi/mediafile/captions/420_v1';
+	const CORTEX_API_GET_METADATA = 'API/v2.2/DataTable/Documents.Video.General-Library-Video:Read';
+	const CORTEX_URL_ASSET_FIELD_VALUE = 'asset-management/[RECORD_ID]?WS=AssetManagement”';
+	const CORTEX_KALTURA_METADATA_FIELD_ID = 'CoreFieldUniqueidentifier';
+	const CORTEX_KALTURA_METADATA_FIELD_CREATED_DATE = 'CoreFieldCreateDate';
+	const CORTEX_KALTURA_METADATA_FIELD_RECORD_ID = 'RecordID';
+	const CORTEX_KALTURA_METADATA_FIELD_PRESENTERS = 'PresentersPersonsPictured';
+	const CORTEX_KALTURA_METADATA_FIELD_JOB_NUMBER = 'JobNumber';
+	/**
+	 * @var string
+	 */
+	private $token;
+	/**
+	 * @var KalturaCortexApiDistributionProfile
+	 */
+	private $distributionProfile;
+	/**
+	 * @var string
+	 */
+	private $cortexSystemId;
+	/**
+	 * @var string
+	 */
+	private $recordId;
+	/**
+	 * @param KalturaDistributionSubmitJobData $data
+	 * @return bool
+	 * @throws Exception
+	 */
+	public function submit(KalturaDistributionSubmitJobData $data)
+	{
+		if(!($data->distributionProfile instanceof KalturaCortexApiDistributionProfile))
+			throw new Exception("Distribution profile must be of type KalturaCortexApiDistributionProfile");
+
+		return $this->doSubmit($data, $data->distributionProfile);
+	}
+	/* (non-PHPdoc)
+	 * @see IDistributionEngineDelete::delete()
+	*/
+	public function delete(KalturaDistributionDeleteJobData $data)
+	{
+		return true;
+	}
+	/**
+	 * @param KalturaDistributionSubmitJobData    $data
+	 * @param KalturaCortexApiDistributionProfile $distributionProfile
+	 * @return bool
+	 * @throws Exception
+	 */
+	protected function doSubmit(KalturaDistributionSubmitJobData $data, KalturaCortexApiDistributionProfile $distributionProfile)
+	{
+		if(!$data->providerData instanceof KalturaCortexApiDistributionJobProviderData)
+		{
+			$this->throwError("Cortex issue: provider data is not KalturaCortexApiDistributionJobProviderData");
+		}
+		$this->setDistributionProfile($distributionProfile);
+		$this->authorizeCortexAccount();
+		$result = $this->uploadVideo($data->providerData);
+		sleep(30);
+		if (kFile::checkFileExists($data->providerData->thumbAssetFilePath))
+		{
+			$this->uploadThumbnail($data->providerData->thumbAssetFilePath);
+		}
+		$this->submitMetadata($data->providerData);
+		if(isset($data->providerData->captionsInfo))
+		{
+			foreach ($data->providerData->captionsInfo as $captionInfo)
+			{
+				/* @var $captionInfo KalturaCortexApiCaptionDistributionInfo */
+				$this->uploadCaption($captionInfo);
```

Comment:
@inbal-ben-david No, because we are not using these assets(caption/thumbnail) after uploading, which is different from the video that we're updating his metadata